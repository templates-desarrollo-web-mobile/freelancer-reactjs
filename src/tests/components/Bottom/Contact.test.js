import Contact from "../../../components/Bottom/Contact";
import { shallow } from 'enzyme';
describe('Pruebas en "Contact"', () => {
    
    const wrapper = shallow(<Contact/>);
    test('Debe de mostrar correctamente el componente', () => {
        
        expect( wrapper ).toMatchSnapshot();
    });

    test('Comprobar si el título es el correcto', () => {
        const title = wrapper.find("h2").text().trim();
        expect(title).toBe("Contact Me");
    });

    test('Comprobar si la fuente de iconos es "fas fa-star"', () => {
        const iconSelect = wrapper.find("i").props().className;
        expect(iconSelect).toBe("fas fa-star");
    });

    test('Comprobar las propiedades de input "name" si son correctas', () => {
        const nameInputValue = wrapper.find("#name");
        const nameInputValueProps = nameInputValue.props();
        expect(nameInputValueProps.placeholder).toBe("Enter your name...");
        expect(nameInputValueProps.type).toBe("text");
        expect(nameInputValueProps.className).toBe("form-control");
    });
    

    /*test('Comprobar si propiedades de imagen son correctas', () => {
        
        const imgData = wrapper.find("img").props();

        expect(imgData.src).toBe(img.url);
        expect(imgData.alt).toBe(img.title);
    });

    test('Comprobar si está añadiendo correctamente clase "animate__fadeIn"', () => {
        
        const divClassName = wrapper.find("div").props().className;
        expect(divClassName.includes('animate__fadeIn')).toBe(true);
    
    });*/
});