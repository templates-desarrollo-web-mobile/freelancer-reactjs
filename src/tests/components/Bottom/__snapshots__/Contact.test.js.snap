// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en "Contact" Debe de mostrar correctamente el componente 1`] = `
<section
  className="page-section"
  id="contact"
>
  <div
    className="container"
  >
    <h2
      className="page-section-heading text-center text-uppercase text-secondary mb-0"
    >
      Contact Me
    </h2>
    <div
      className="divider-custom"
    >
      <div
        className="divider-custom-line"
      />
      <div
        className="divider-custom-icon"
      >
        <i
          className="fas fa-star"
        />
      </div>
      <div
        className="divider-custom-line"
      />
    </div>
    <div
      className="row justify-content-center"
    >
      <div
        className="col-lg-8 col-xl-7"
      >
        <form
          data-sb-form-api-token="API_TOKEN"
          id="contactForm"
        >
          <div
            className="form-floating mb-3"
          >
            <input
              className="form-control"
              data-sb-validations="required"
              id="name"
              placeholder="Enter your name..."
              type="text"
            />
            <label
              for="name"
            >
              Full name
            </label>
            <div
              className="invalid-feedback"
              data-sb-feedback="name:required"
            >
              A name is required.
            </div>
          </div>
          <div
            className="form-floating mb-3"
          >
            <input
              className="form-control"
              data-sb-validations="required,email"
              id="email"
              placeholder="name@example.com"
              type="email"
            />
            <label
              for="email"
            >
              Email address
            </label>
            <div
              className="invalid-feedback"
              data-sb-feedback="email:required"
            >
              An email is required.
            </div>
            <div
              className="invalid-feedback"
              data-sb-feedback="email:email"
            >
              Email is not valid.
            </div>
          </div>
          <div
            className="form-floating mb-3"
          >
            <input
              className="form-control"
              data-sb-validations="required"
              id="phone"
              placeholder="(123) 456-7890"
              type="tel"
            />
            <label
              for="phone"
            >
              Phone number
            </label>
            <div
              className="invalid-feedback"
              data-sb-feedback="phone:required"
            >
              A phone number is required.
            </div>
          </div>
          <div
            className="form-floating mb-3"
          >
            <textarea
              className="form-control"
              data-sb-validations="required"
              id="message"
              placeholder="Enter your message here..."
              style={
                Object {
                  "height": "10rem",
                }
              }
              type="text"
            />
            <label
              for="message"
            >
              Message
            </label>
            <div
              className="invalid-feedback"
              data-sb-feedback="message:required"
            >
              A message is required.
            </div>
          </div>
          <div
            className="d-none"
            id="submitSuccessMessage"
          >
            <div
              className="text-center mb-3"
            >
              <div
                className="fw-bolder"
              >
                Form submission successful!
              </div>
              To activate this form, sign up at
              <br />
              <a
                href="https://startbootstrap.com/solution/contact-forms"
              >
                https://startbootstrap.com/solution/contact-forms
              </a>
            </div>
          </div>
          <div
            className="d-none"
            id="submitErrorMessage"
          >
            <div
              className="text-center text-danger mb-3"
            >
              Error sending message!
            </div>
          </div>
          <button
            className="btn btn-primary btn-xl disabled"
            id="submitButton"
            type="submit"
          >
            Send
          </button>
        </form>
      </div>
    </div>
  </div>
</section>
`;
